analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContext.cs

```diff
@@ -18,239 +18,21 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-using System.IO;
-using System.Runtime.CompilerServices;
-using System.Text.RegularExpressions;
-using System.Xml.Linq;
 using Microsoft.CodeAnalysis.Text;
-using static SonarAnalyzer.Helpers.DiagnosticDescriptorFactory;
 
 namespace SonarAnalyzer;
 
-/// <summary>
-/// SonarC# and SonarVB specific context for initializing an analyzer. This type acts as a wrapper around Roslyn
-/// <see cref="AnalysisContext"/> to allow for specialized control over the analyzer.
-/// Here is the list of fine-grained changes we are doing:
-/// - Avoid duplicated issues when the analyzer NuGet (SonarAnalyzer) and the VSIX (SonarLint) are installed simultaneously.
-/// - Allow a specific kind of rule-set for SonarLint (enable/disable a rule).
-/// - Prevent reporting an issue when it was suppressed on SonarQube.
-/// </summary>
-public class SonarAnalysisContext
+public partial /*FIXME: REMOVE partial */ class SonarAnalysisContext : SonarAnalysisContextBase
```

Comment:
These commits are very hard to review. When you change something like `file scoped namespaces` please do not combine it with other changes like namespace removal, adding comments, changing visibility for some classes.

Reply:
I'm sorry, there were too many things shipped in that commit